script.module.codequick/lib/codequick/support.py

```diff
@@ -86,52 +83,93 @@ def emit(self, record):
             xbmc.log("###### debug ######", xbmc.LOGWARNING)
 
 
+class Gettext(object):
+    def __init__(self):
+        self._loaded = False
+        self._strings = {}
+
+    def gettext(self, string_id):  # type: (str) -> str
+        if string_id in self._strings or (self._loaded is False and string_id in self._parse_po()):
+            numeric_id = self._strings[string_id]
+            return addon_data.getLocalizedString(numeric_id)
+        else:
+            raise KeyError("no localization found for string id '%s'" % string_id)
+
+    def _parse_po(self):  # type: () -> dict
+        self._loaded = True
+
+        # Check if path needs to be translated first
+        addon_path = addon_data.getAddonInfo("path")
+        if addon_path[:10] == "special://":  # pragma: no cover
+            addon_path = xbmc.translatePath(addon_path)
+
+        # Location of strings.po file
+        strings_po = os.path.join(addon_path, "resources", "language", "resource.language.en_gb", "strings.po")
```

Comment:
you're not allowed to directly parse/mess with those files. Use the API only please. I know you're trying to add support for strings instead of numeric id's but nevertheless, you need to use another approach that does not rely on reading the strings.po file directly